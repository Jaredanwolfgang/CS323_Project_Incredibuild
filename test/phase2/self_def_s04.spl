int main() {
    int x = 42;       // Declare an integer variable
    int *ptr = &x;    

    float y = 30.0;
    float *ptr_f = &x;
    float *ptr_t = &y;

    char z = 'a';
    char *ptr_z = z;
}