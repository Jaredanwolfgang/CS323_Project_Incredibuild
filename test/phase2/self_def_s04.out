[Semantic Error] Invalid Operation Error[5] at line 6: "Assigning a value of type Primitive: type: Pointer(Int) to a variable of type Primitive: type: Pointer(Float)"
[Semantic Error] Invalid Operation Error[5] at line 10: "Assigning a value of type Primitive: type: Char to a variable of type Primitive: type: Pointer(Char)"